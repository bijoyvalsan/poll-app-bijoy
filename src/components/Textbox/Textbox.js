import React from "react";
import PropTypes from "prop-types";

import { StyledContainer, StyledLabelText } from "./Textbox.styled";

const Textbox = ({ placeholder, onChange, value, label, id }) => {
  return (
    <>
      {label && <StyledLabelText>{label}</StyledLabelText>}
      <StyledContainer>
        <input
          id={id}
          data-testid={id}
          aria-label={placeholder}
          onChange={onChange}
          tabIndex="0"
          type="text"
          placeholder={placeholder}
          value={value}
        />
      </StyledContainer>
    </>
  );
};

Textbox.defaultProps = {
  placeholder: "search",
  value: "",
  label: undefined
};

Textbox.propTypes = {
  id: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  value: PropTypes.string,
  label: PropTypes.string,
  onChange: PropTypes.func.isRequired
};

export default React.memo(Textbox);
