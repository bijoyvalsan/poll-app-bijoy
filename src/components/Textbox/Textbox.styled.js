import styled from "styled-components";

export const StyledLabelText = styled.p`
  margin: 0 0 12px 0;
`;

export const StyledContainer = styled.div`
  position: relative;
  margin-bottom: 24px;
  input[type="text"] {
    appearance: none;
    background-color: white;
    border: none;
    outline: 0px;
    border-radius: 8px;
    padding: 12px;
    width: 100%;
    margin: 0px;
    font-size: 16px;
    line-height: 24px;
    box-shadow: rgb(153 153 153) 0px 0px 0px 1px;
    box-sizing: border-box;
  }
  input[type="text"]:focus {
    box-shadow: #3388ff 0px 0px 0px 2px;
  }
`;
