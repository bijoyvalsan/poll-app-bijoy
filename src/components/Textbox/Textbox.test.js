import React from "react";
import { render } from "@testing-library/react";

import Textbox from "./Textbox";

describe("Textbox Component", () => {
  it("should render with default Textbox styles", () => {
    const mockData = {
      id: "txt-search",
      placeholder: "search",
      value: "",
      label: "search",
      onChange: jest.fn()
    };
    const { asFragment } = render(<Textbox {...mockData} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
