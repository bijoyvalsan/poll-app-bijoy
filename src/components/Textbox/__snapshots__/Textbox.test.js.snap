// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textbox Component should render with default Textbox styles 1`] = `
<DocumentFragment>
  <p
    class="sc-bdnxRM emSMCL"
  >
    search
  </p>
  <div
    class="sc-gtsrHT fGikmx"
  >
    <input
      aria-label="search"
      data-testid="txt-search"
      id="txt-search"
      placeholder="search"
      tabindex="0"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
